### PR TITLE
fixe for proxy

### PR DESCRIPTION
**Pull request by FrankCRoth**

### CODE REVIEW

File: lib/azure_jwt_auth/authenticable.rb [hunk 1]
```diff
@@ -16,16 +17,16 @@ def authenticate!
       unauthorize! unless JwtManager.providers
 
       JwtManager.providers.each do |_uid, provider|
-        token = JwtManager.new(request, provider.uid)
-
-        if token.valid?
-          @current_user = entity_from_token_payload(token.payload)
-          break
+	begin
```
Comment (rjurado01) [inline lib/azure_jwt_auth/authenticable.rb]: I don't understand what problem are you fixed here.

File: lib/azure_jwt_auth/authenticable.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@ module AzureJwtAuth
   AzureJwtAuth::NotAuthorized = Class.new(StandardError)
 
   module Authenticable
+
```
Comment (rjurado01) [inline lib/azure_jwt_auth/authenticable.rb]: This blank line doesn't follow the style of this gem. We don't add empty line at begining of modules.

File: lib/azure_jwt_auth/provider.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-require 'net/http'
+require 'net/https'
```
Comment (rjurado01) [inline lib/azure_jwt_auth/provider.rb]: This is not needed. From [Https section of documentation](https://ruby-doc.org/stdlib-2.6.5/libdoc/net/http/rdoc/Net/HTTP.html):

```
In previous versions of Ruby you would need to require 'net/https' to use HTTPS. This is no longer true.
```

**Comment by rjurado01** [comment]
Thanks for your work but this pull request brings no value and it hasn't any test to support changes.

If you add new changes please reopen it.

**Comment by FrankCRoth** [comment]
Hi,

Not sure how to test proxy but for sure needs to be added in order to work with proxy.

**Comment by FrankCRoth** [comment]
> On 3 Oct 2019, at 13:38, Rafael Jurado González <notifications@github.com> wrote:
> 
> @rjurado01 commented on this pull request.
> 
> In lib/azure_jwt_auth/authenticable.rb <https://github.com/nosolosoftware/azure_jwt_auth/pull/2#discussion_r330989831>:
> 
> > @@ -16,16 +17,16 @@ def authenticate!
>        unauthorize! unless JwtManager.providers
>  
>        JwtManager.providers.each do |_uid, provider|
> -        token = JwtManager.new(request, provider.uid)
> -
> -        if token.valid?
> -          @current_user = entity_from_token_payload(token.payload)
> -          break
> +	begin
> I don't understand what problem are you fixed here.
> 
> 


The code did not run at all that is why I had to add the begin statement. Was missing before.


> 
> In lib/azure_jwt_auth/authenticable.rb <https://github.com/nosolosoftware/azure_jwt_auth/pull/2#discussion_r330991619>:
> 
> > @@ -4,6 +4,7 @@ module AzureJwtAuth
>    AzureJwtAuth::NotAuthorized = Class.new(StandardError)
>  
>    module Authenticable
> +
> This blank line doesn't follow the style of this gem. We don't add empty line at begining of modules.
> 
> 


Sure, just remove ;-)

> 
> In lib/azure_jwt_auth/provider.rb <https://github.com/nosolosoftware/azure_jwt_auth/pull/2#discussion_r330993469>:
> 
> > @@ -1,4 +1,4 @@
> -require 'net/http'
> +require 'net/https'
> This is not needed. From Https section of documentation <https://ruby-doc.org/stdlib-2.6.5/libdoc/net/http/rdoc/Net/HTTP.html>:
> 
> In previous versions of Ruby you would need to require 'net/https' to use HTTPS. This is no longer true.

Did not work for me. I needed to change to https in order to get it to work correctly with proxy. The version provided did not recognise proxy settings at all and as we are using proxy in our company the change was required.

**Comment by rjurado01** [comment]
Ok. Can you make a new pull request only with this code?

**Pull request closed without merging**